src/sentry/web/frontend/accounts.py

```diff
@@ -187,11 +187,19 @@ def account_settings(request):
 
         form.save()
 
-        # remove previously valid email address
+        # update notification settings for those set to primary email with new primary email
+        alert_email = UserOption.objects.get_value(user=user, project=None, key='alert_email')
+
+        if alert_email == old_email:
+            UserOption.objects.set_value(user=user, project=None, key='alert_email', value=user.email)
+        options = UserOption.objects.filter(user=user, key='mail:email')
+        for option in [o for o in options if o.value == old_email]:
```

Comment:
I won't block for this, but this would be an odd pattern in python and it incurs two loops.

I'd spell it out as:

```python
for option in options:
    if o.value != old_email:
        continue
    option.value = user.email
    option.save()
```